app/controllers/carts_controller.rb

```diff
@@ -0,0 +1,70 @@
+class CartsController < ApplicationController
+  before_action :set_cart, only: %i[ show edit update destroy ]
+
+  # GET /carts or /carts.json
+  def index
+    @carts = Cart.all
+  end
+
+  # GET /carts/1 or /carts/1.json
+  def show
+  end
+
+  # GET /carts/new
+  def new
+    @cart = Cart.new
+  end
+
+  # GET /carts/1/edit
+  def edit
+  end
+
+  # POST /carts or /carts.json
+  def create
+    @cart = Cart.new(cart_params)
+
+    respond_to do |format|
+      if @cart.save
+        format.html { redirect_to cart_url(@cart), notice: "Cart was successfully created." }
+        format.json { render :show, status: :created, location: @cart }
+      else
+        format.html { render :new, status: :unprocessable_entity }
+        format.json { render json: @cart.errors, status: :unprocessable_entity }
+      end
+    end
+  end
+
+  # PATCH/PUT /carts/1 or /carts/1.json
+  def update
+    respond_to do |format|
+      if @cart.update(cart_params)
+        format.html { redirect_to cart_url(@cart), notice: "Cart was successfully updated." }
+        format.json { render :show, status: :ok, location: @cart }
+      else
+        format.html { render :edit, status: :unprocessable_entity }
+        format.json { render json: @cart.errors, status: :unprocessable_entity }
+      end
+    end
+  end
+
+  # DELETE /carts/1 or /carts/1.json
+  def destroy
+    @cart.destroy
+
```

Comment:
Consider checking that the cart id matches the session id as an added check before deleting the cart.
```suggestion
  @cart. destroy if @cart.id == session[:cart_id]
  @session[:cart_id] = nil
```